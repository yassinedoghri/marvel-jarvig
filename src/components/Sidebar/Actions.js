import styled from "styled-components";

const Actions = styled.nav`
  color: white;
  font-family: Bangers;
  text-align: center;
  letter-spacing: 0.1em;
`;

export default Actions;
