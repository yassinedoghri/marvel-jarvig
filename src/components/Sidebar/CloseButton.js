import styled from "styled-components";

const CloseButton = styled.button`
  position: absolute;
  background-color: transparent;
  color: ${props => props.theme.colors.contrastDark.secondary};
  border: none;
  top: 0;
  right: 0;
  font-size: 3em;

  &:hover {
    cursor: pointer;
    color: ${props => props.theme.colors.contrastDark.primary};
  }
`;

export default CloseButton;
