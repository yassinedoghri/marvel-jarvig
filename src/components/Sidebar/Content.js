import styled from "styled-components";

const Content = styled.div`
  padding: ${props => props.theme.spaces.lg}em;
  text-align: justify;
  color: ${props => props.theme.colors.contrastDark.base};
`;

export default Content;
