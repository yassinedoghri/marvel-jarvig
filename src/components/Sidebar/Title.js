import styled from "styled-components";

const Title = styled.h2`
  font-family: Bangers, sans-serif;
  padding-left: ${props => props.theme.spaces.md}em;
  color: white;
  letter-spacing: 0.1em;
`;

export default Title;
