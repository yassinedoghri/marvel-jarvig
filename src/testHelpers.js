import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import theme from "theme";
import { ThemeProvider } from "styled-components";

export default function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}
