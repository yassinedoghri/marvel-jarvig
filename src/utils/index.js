export { default as style } from "utils/style";
export { default as baseStyles } from "utils/baseStyles";
export { default as helpers } from "utils/helpers";
export { default as testHelpers } from "utils/testHelpers";
export { default as registerServiceWorker } from "utils/registerServiceWorker";
export { default as media } from "utils/media";
