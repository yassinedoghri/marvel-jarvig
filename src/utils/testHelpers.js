import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import {
  ConnectedRouter,
  routerMiddleware,
  routerReducer
} from "react-router-redux";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import rootReducer from "reducers";
import { applyMiddleware, combineReducers, createStore } from "redux";
import { ThemeProvider } from "styled-components";
import theme from "theme";

const history = createMemoryHistory("/");
const routeMiddleware = routerMiddleware(history);

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  applyMiddleware(routeMiddleware)
);

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export function renderInContext(component) {
  return renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}

export function ShallowRenderInContext(component) {
  const renderer = new ShallowRenderer();
  return renderer.render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}
