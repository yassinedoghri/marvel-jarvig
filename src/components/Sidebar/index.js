import CloseButton from "components/Sidebar/CloseButton";
import Content from "components/Sidebar/Content";

import Title from "components/Sidebar/Title";
import { transitions } from "polished";
import styled from "styled-components";
import media from "utils/media";

const Sidebar = styled.aside`
  background-color: ${props => props.theme.colors.foreground.base};
  color: ${props => props.theme.colors.contrastDark.primary};
  left: ${props => (props.open ? "0" : "-300px")};
  box-shadow: 0 3px 15px rgba(0, 0, 0, 0.2);
  width: 100%;
  position: fixed;
  height: 100%;
  top: 0;
  left: ${props => (props.open ? "0" : "-100%")};
  overflow-y: auto;
  z-index: 3;
  padding-bottom: 3em;
  ${transitions("all 500ms ease-in-out")};

  ${media.tablet`
    width: 20em;
  `};
`;

Sidebar.Title = Title;
Sidebar.Content = Content;
Sidebar.CloseButton = CloseButton;

export default Sidebar;
