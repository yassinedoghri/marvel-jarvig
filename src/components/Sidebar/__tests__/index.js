import Sidebar from "components/Sidebar";
import React from "react";
import { renderWithTheme } from "utils/testHelpers";

describe("Sidebar Block", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      <Sidebar>
        <Sidebar.Title>Sidebar Title</Sidebar.Title>
        <Sidebar.CloseButton>X</Sidebar.CloseButton>
        <Sidebar.Content>This is a Sidebar</Sidebar.Content>
      </Sidebar>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("adds open modifier", () => {
    const tree = renderWithTheme(
      <Sidebar open>
        <Sidebar.Title>Sidebar Title</Sidebar.Title>
        <Sidebar.CloseButton>X</Sidebar.CloseButton>
        <Sidebar.Content>This is a Sidebar</Sidebar.Content>
      </Sidebar>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
