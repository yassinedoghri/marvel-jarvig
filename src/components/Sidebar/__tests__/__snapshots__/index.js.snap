// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Block adds open modifier 1`] = `
<aside
  className="sc-bxivhb kVpWcR"
  open={true}
>
  <h2
    className="sc-htpNat bWyIsR"
  >
    Sidebar Title
  </h2>
  <button
    className="sc-bdVaJa gAPlTG"
  >
    X
  </button>
  <div
    className="sc-bwzfXH jSvnsT"
  >
    This is a Sidebar
  </div>
</aside>
`;

exports[`Sidebar Block renders correctly 1`] = `
<aside
  className="sc-bxivhb hOqVLK"
>
  <h2
    className="sc-htpNat bWyIsR"
  >
    Sidebar Title
  </h2>
  <button
    className="sc-bdVaJa gAPlTG"
  >
    X
  </button>
  <div
    className="sc-bwzfXH jSvnsT"
  >
    This is a Sidebar
  </div>
</aside>
`;
